### ISSUE TITLE
[openssl] build failure on x64-windows-static

### ISSUE THREAD

**Issue by Dragonrun1** [microsoft/vcpkg#14963]
**Host Environment**
- OS: Win10
- Compiler: revision

**To Reproduce**
`.\vcpkg.exe --triplet x64-windows-static install openssl`

**Failure logs** 
```
Computing installation plan...
The following packages will be built and installed:
    openssl[core]:x64-windows-static -> 1.1.1h#3
Detecting compiler hash for triplet x64-windows-static...
Could not locate cached archive: C:\Users\mgcum\AppData\Local\vcpkg\archives\16\169383a1f6432196eefc839d0085ca7c5d72099c.zip
Starting package 1/1: openssl:x64-windows-static
Building package openssl[core]:x64-windows-static...
-- Using cached C:/vcpkg/vcpkg/downloads/openssl-1.1.1h.tar.gz
-- Cleaning sources at C:/vcpkg/vcpkg/buildtrees/openssl/src/openssl-1-09fb822c56.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/vcpkg/vcpkg/downloads/openssl-1.1.1h.tar.gz
-- Using source at C:/vcpkg/vcpkg/buildtrees/openssl/src/openssl-1-09fb822c56.clean
-- Copying openssl release source files...
-- Copying openssl release source files... done
-- Configure x64-windows-static-rel
-- Configure x64-windows-static-rel done
-- Build x64-windows-static-rel
-- Build x64-windows-static-rel done
-- Copying openssl debug source files...
-- Copying openssl debug source files... done
-- Configure x64-windows-static-dbg
-- Configure x64-windows-static-dbg done
-- Build x64-windows-static-dbg
-- Build x64-windows-static-dbg done
CMake Error at scripts/cmake/vcpkg_copy_tool_dependencies.cmake:26 (message):
  Could not find PowerShell Core; please open an issue to report this.
Call Stack (most recent call first):
  ports/openssl/windows/portfile.cmake:150 (vcpkg_copy_tool_dependencies)
  ports/openssl/portfile.cmake:19 (include)
  scripts/ports.cmake:136 (include)


Error: Building package openssl:x64-windows-static failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: openssl:x64-windows-static
  Vcpkg version: 2020.11.12-nohash

Additionally, attach any relevant sections from the log files above.
```
**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

**Comment by StarGate-One** [comment]
>-- Build x64-windows-static-dbg done
>CMake Error at scripts/cmake/vcpkg_copy_tool_dependencies.cmake:26 (message):
**>Could not find PowerShell Core; please open an issue to report this.**
>Call Stack (most recent call first):

1. When is the last time vcpkg was updated as many changes to the core program (vcpkg.exe) has happened the past few days?
`git pull` - get the latest vcpkg changes
`.\bootstrap-vcpkg.bat` - to rebuild the vcpkg binary using updated sources (only needed when `[vcpkg-root]\toolsrc` files are add, changed and/or deleted)
`vcpkg upgrade` - to see if any existing packages need upgrading
2. I just created a fresh new clone and was able to build without any problems?
>->.\vcpkg.exe --triplet x64-windows-static install openssl
>Computing installation plan...
>The following packages will be built and installed:
>    openssl[core]:x64-windows-static -> 1.1.1h#3
>Starting package 1/1: openssl:x64-windows-static
>Building package openssl[core]:x64-windows-static...
>-- Downloading https://www.openssl.org/source/openssl-1.1.1h.tar.gz...
>-- Downloading https://strawberry.perl.bot/download/5.30.0.1/strawberry-perl-5.30.0.1-32bit.zip...
>-- Downloading https://strawberry.perl.bot/download/5.30.0.1/strawberry-perl-5.30.0.1-32bit.zip... Failed. Status: 22;"HTTP response code said error"
>-- Downloading http://strawberryperl.com/download/5.30.0.1/strawberry-perl-5.30.0.1-32bit.zip...
>-- Extracting source D:/Git/vcpkg/test/downloads/openssl-1.1.1h.tar.gz
>-- Using source at D:/Git/vcpkg/test/buildtrees/openssl/src/openssl-1-09fb822c56.clean
>-- Downloading https://www.nasm.us/pub/nasm/releasebuilds/2.14.02/win32/nasm-2.14.02-win32.zip...
>-- Downloading https://download.qt.io/official_releases/jom/jom_1_1_3.zip...
>-- Copying openssl release source files...
>-- Copying openssl release source files... done
>-- Configure x64-windows-static-rel
>-- Configure x64-windows-static-rel done
>-- Build x64-windows-static-rel
>-- Build x64-windows-static-rel done
>-- Copying openssl debug source files...
>-- Copying openssl debug source files... done
>-- Configure x64-windows-static-dbg
>-- Configure x64-windows-static-dbg done
>-- Build x64-windows-static-dbg
>-- Build x64-windows-static-dbg done
>-- Installing: D:/Git/vcpkg/test/packages/openssl_x64-windows-static/share/openssl/copyright
>-- Installing: D:/Git/vcpkg/test/packages/openssl_x64-windows-static/share/openssl//usage
>-- Performing post-build validation
>-- Performing post-build validation done
>Building package openssl[core]:x64-windows-static... done
>Installing package openssl[core]:x64-windows-static...
>Installing package openssl[core]:x64-windows-static... done
>Elapsed time for package openssl:x64-windows-static: 1.423 min
>
>Total elapsed time: 1.431 min
>
>The package openssl is compatible with built-in CMake targets:
>
>    find_package(OpenSSL REQUIRED)
>    target_link_libraries(main PRIVATE OpenSSL::SSL OpenSSL::Crypto)

**Comment by Dragonrun1** [comment]
Just installed it a few hours ago so I'm pretty sure I have the latest version. Actual just grabbed it again like 2 hours ago to try build again after wiping directory and still doesn't build

**Comment by sfhacker** [comment]
Are we **STILL** having issues building OpenSSL from source on Windows?

![image](https://user-images.githubusercontent.com/25817529/101280398-eceb1080-381c-11eb-9783-4869fcfe1b48.png)

> Could not find PowerShell Core; please open an issue to report this.

Why you need PowerShell to build OpenSSL? I'm pretty sure I built OpenSSL on Windows without PowerShell!
@Dragonrun1 Have you tried other alternatives? (e.g. nmake, etc)

**Comment by Dragonrun1** [comment]
I have no idea why vcpkg would need PowerShell to build on Windows I just know for some reason someone decided it does in vcpkg.

So some more background on what I'm doing when I ran into the bug. I'm trying to install in rust's cargo a diesel-cli tool which uses a crate called mysqlclient-sys which looks for a static build of the mysql client library. One way it tries to find it is by using vcpkg. vcpkg when you do `vcpkg.exe --triplet x64-windows-static install` requires openssl to be built which is now broken thus this issue.

I've also done a little testing and this also break stuff for the PostgreSQL build as well since it also require openssl as part of it's build which is not surprising since almost everything in open source uses openssl for anything SSL related.

I notice the version of openssl you have is actually outdate compared to what vcpkg tries to build which is the 1.1.1h version.

**So to sum it up have OpenSSL not working in vcpkg breaks almost everything security related in Windows that is trying to use it again!**

**Comment by StarGate-One** [comment]
1. I would try:
   - delete file `[vcpkg-root]\downloads\PowerShell-7.0.3-win-x86.zip`
   - deleting directory `[vcpkg-root]\downloads\tools\powershell-core-7.0.3-windows`
   - install openssl again
2. This should force vcpkg to download/install powershell core again.

**Comment by Dragonrun1** [comment]
Actuallyu right now neither exists on my system but I was trying some other ideas I found floating around the Internet so I'll for the 4-5? time delete my vcpkg root directory and start over. It'll only take another couple hours to see if it helps ;)

**Comment by Dragonrun1** [comment]
So everything downloaded faster today for me but OpenSSL failed again and there is nothing in downloads directory with PowerShell in it's name so that might be why it can't find it ;)

**Comment by StarGate-One** [comment]
1. Is there a powershell directory in your PATH env variable?
2. `C:\Windows\System32\WindowsPowerShell\v1.0\` should be there, any others should be removed, if the directory does not exist.
3. Try downloading it from https://github.com/PowerShell/PowerShell/releases/download/v7.0.3/PowerShell-7.0.3-win-x86.zip and putting the file in `[vcpkg-root]\downloads` - make sure to name the file `PowerShell-7.0.3-win-x86.zip`

**Comment by vicroms** [comment]
Vcpkg should download a copy of PowerShell Core during first use of `vcpkg install`, e.g.:

```
A suitable version of powershell-core was not found (required v7.0.3). Downloading portable powershell-core v7.0.3...
Downloading powershell-core...
  https://github.com/PowerShell/PowerShell/releases/download/v7.0.3/PowerShell-7.0.3-win-x86.zip -> E:\Dev\Source\vcpkg\downloads\PowerShell-7.0.3-win-x86.zip
Extracting powershell-core...
```
Do you see that message when running `vcpkg install openssl`? 

PowerShell Core is required by `vcpkg_copy_tool_dependencies()` in order to run `applocal.ps1`. OpenSSL produces an executable `OpenSSL.exe`, the `applocal.ps1`script detects which `.dll` are required for the executable to run and copies those DLLs to the same directory as the executable.

If you don't have a need for `OpenSSL.exe` produced by the OpenSSL port, a quick patch would be to comment the call to `vcpkg_copy_tool_dependencies()` from `ports/openssl/windows/portfile.cmake`.

**Comment by Dragonrun1** [comment]
C:\Windows\System32\WindowsPowerShell\v1.0\ is where it's finding PS5 which was causing the problem.

**Comment by JackBoosY** [comment]
Another issue with `vcpkg_copy_tool_dependencies`...
@ras0219 @BillyONeal Please consider to fix this issue first.

Duplicated to #12849.
